### PR TITLE
Add kerberos support to `anaconda login`

### PR DESCRIPTION
**Pull request by dsludwig**
Related to Anaconda-Server/Anaconda-server#1597 Anaconda-Server/Anaconda-server#1566

### CODE REVIEW

**Comment by dsludwig** [comment]
@srossross how do you think users should interact with kerberos from the command line?

I have added a flag that tries to do the entire authentication flow.

i.e.
```
$ anaconda login --kerberos
Login successful
```

File: binstar_client/commands/login.py [hunk 1]
```diff
@@ -118,4 +153,6 @@ def add_parser(subparsers):
                            help="Specify your password. "
                                 "If this is not given, you will be prompted"
                            )
+    subparser.add_argument('--kerberos', action='store_true',
```
Comment (srossross) [inline binstar_client/commands/login.py]: This should probably be a config variable. 

Also, could the client this with the server? 

e.g. a pre step: `[GET] api/auth/authentication-type`

Comment (dsludwig) [inline binstar_client/commands/login.py]: Yeah, I was considering setting it as a configuration variable, just started with argument for ease of development.

I think the preflight request is the best option - you shouldn't be allowed to input username/password to a Kerberos-configured server.

**Comment by dsludwig** [comment]
@srossross are there any other things I can improve here?

**Comment by srossross** [comment]
:+1:

**Pull request merged**